text/0486-deprecate-mouseenter.md

```diff
@@ -0,0 +1,162 @@
+not a single component/action needs them.  
+
+Deprecating support for `mousemove` is also proposed, which is a (bubbling) event that does not have the higher 
+implementation cost as `mouseenter`/`mouseleave`, but nevertheless requires the EventDispatcher to optimistically handle
+these extremely high-volume events.
+
+While efforts to make this more "pay as you go" are [possible](https://github.com/emberjs/ember.js/pull/17911), 
+the trade-off of keeping support around still seems unfavorable, as these events fire so
+frequently, while they are (most certainly) very rarely used.
+
+This is even more so given that Glimmer Components with their outerHTML semantics do not 
+work with event handler methods, and `{{action}}` will eventually fade away in favor of
+`{{on}}` using native `addListener()`.
+
+## Transition Path
+
+When a `mouseenter`/`mouseleave`/`mousemove` event is dispatched by the EventDispatcher to a component
+or `{{action}}` listening to that event type, a deprecation warning will be issued.
+
+The linked migration guide will cover these use cases:
+
+#### Ember.Component
+
+Before:
+
+```js
+import Component from '@ember/component';
+
+export default class MyComponent extends Component {
+  mouseEnter(e) {
+    // do something
+  }
+}
+```
+
+After:
+
+```js
+import Component from '@ember/component';
+import { action } from '@ember/object';
+
+export default class MyComponent extends Component {
+  @action
+  handleMouseEnter(e) {
+    // do something
+  }
+  
+  didInsertElement() {
+    super.didInsertElement(...arguments);
+    this.element.addEventListener('mouseenter', this.handleMouseEnter);
+  }
+  
+  willDestroyElement() {
+    super.willDestroyElement(...arguments);
+    this.element.removeEventListener('mouseenter', this.handleMouseEnter);
+  }
+}
+```
+
+An alternative to attaching the event listener in the component class is to opt into outer HTML semantics by making the
+component tag-less and using the `{{on}}` modifier in the template:
+
+```js
+import Component from '@ember/component';
+import { action } from '@ember/object';
+
+export default class MyComponent extends Component {
+  tagName = '';
+  
+  @action
+  handleMouseEnter(e) {
+    // do something
+  }
+}
+```
+
+```hbs
+<div {{on "mouseenter" this.handleMouseEnter}}>
+  ...
+</div>
+```
+
+#### `{{action}}` modifier
+
+Before:
+
+```hbs
+<button {{action "handleMouseEnter" on="mouseEnter"}}>Hover</button>
+```
+
+After (based on [RFC471](https://github.com/emberjs/rfcs/blob/master/text/0471-on-modifier.md)):
+
+```hbs
+<button {{on "mouseenter" this.handleMouseEnter}}>Hover</button>
+```
+
+## How We Teach This
+
+The deprecation guide should explain the transition path as shown above.
+
+The Guide's [Handling Events](https://guides.emberjs.com/release/components/handling-events/#toc_event-names)
+section must remove the reference to `mouseenter`, `mouseleave` and `mousemove`.
```

Comment:
```suggestion
section must remove the reference to `mouseenter`, `mouseleave` and `mousemove`. It should also be removed from the API docs for [Components events](https://api.emberjs.com/ember/release/classes/Component) and [event delegation](https://api.emberjs.com/ember/release/classes/Application)
```

Please double check whether these actually need to change. I found them by searching the ember.js codebase for `mouseenter`. A similar search should be done for mouseleave and mousemove in case they are used in examples in other parts of the API docs.

Reply:
Good catch, thanks! The references for `Component` should be removed, while the example for `Application` shows a way to opt out from listening to those events, so that looks valid and could maybe stay in place until 4.0?

I haven't found any other references in the API docs...